libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -50,14 +51,19 @@ struct CopyTranslate: public yul::ASTCopier
 {
 	using ExternalRefsMap = std::map<yul::Identifier const*, InlineAssemblyAnnotation::ExternalIdentifierInfo>;
 
-	CopyTranslate(IRGenerationContext& _context, ExternalRefsMap const& _references):
-		m_context(_context), m_references(_references) {}
+	CopyTranslate(yul::Dialect const& _dialect, IRGenerationContext& _context, ExternalRefsMap const& _references):
+		m_dialect(_dialect), m_context(_context), m_references(_references) {}
 
 	using ASTCopier::operator();
 
 	yul::YulString translateIdentifier(yul::YulString _name) override
 	{
-		return yul::YulString{"usr$" + _name.str()};
+		// Strictly, we should check that this builtin also exists in the target dialect,
+		// but currently, eveything is EVM anyway.
+		if (m_dialect.builtin(_name))
```

Comment:
Added this.

Reply:
Is this what the comment says it doesn't do currently?

Reply:
What I was referring to was that the inline assembly snippet could use a different dialect than the dialect we are currently compiling to. So the current code basically assumes that all built-in functions in the assembly snippet also exist in the target Yul dialect and have the same name.

Reply:
ah ok